File: sdkwork-ai-sdk-python/sdkwork_ai_sdk/api/audio2.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import CreateTranscriptionRequest, CreateTranslationPostRequest, SpeechRequest, TranscriptionResponse

class Audio2Api:
    """Audio API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def create_translation(self, body: Optional[CreateTranslationPostRequest] = None, params: Optional[Dict[str, Any]] = None) -> TranscriptionResponse:
        """Create translation"""
        return self._client.post(f"/ai/v3/v1/audio/translations", data=body, params=params)

    def create_transcription(self, body: Optional[CreateTranscriptionRequest] = None, params: Optional[Dict[str, Any]] = None) -> TranscriptionResponse:
        """Create transcription"""
        return self._client.post(f"/ai/v3/v1/audio/transcriptions", data=body, params=params)

    def create_speech(self, body: SpeechRequest) -> str:
        """Create speech"""
        return self._client.post(f"/ai/v3/v1/audio/speech", json=body)

File: sdkwork-ai-sdk-flutter/lib/src/api/audio2.dart
import '../http/client.dart';
import '../models.dart';

class Audio2Api {
  final HttpClient _client;
  
  Audio2Api(this._client);

  /// Create translation
  Future<TranscriptionResponse?> createTranslation(CreateTranslationPostRequest? body, Map<String, dynamic>? params) async {
    final response = await _client.post(ApiPaths.aiPath('/v1/audio/translations'), body: body, params: params, contentType: 'multipart/form-data');
    return response is TranscriptionResponse ? response : null;
  }

  /// Create transcription
  Future<TranscriptionResponse?> createTranscription(CreateTranscriptionRequest? body, Map<String, dynamic>? params) async {
    final response = await _client.post(ApiPaths.aiPath('/v1/audio/transcriptions'), body: body, params: params, contentType: 'multipart/form-data');
    return response is TranscriptionResponse ? response : null;
  }

  /// Create speech
  Future<String?> createSpeech(SpeechRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/v1/audio/speech'), body: body, contentType: 'application/json');
    return response is String ? response : null;
  }
}
